src/app/ReadHandler.cpp

```diff
@@ -284,6 +284,11 @@ CHIP_ERROR ReadHandler::ProcessAttributePathList(AttributePathList::Parser & aAt
         }
         SuccessOrExit(err);
 
+        if (MergeOverlappedAttributePath(clusterInfo))
+        {
+            continue;
+        }
```

Comment:
the PushFront below does more:
   - SetDirty()
   - SetInitialReport()
   
Why are these not needed if merge is possible?
In this case, it may be prefferable to do if/else instead of continue to make intent clear that there are two paths to add the new cluster  info: one is to merge due to overlap, another is to append to the end.

To a reader it seems that both paths should have the same  setdirty/sendinitial logic. If this is not the case, some comments should explain why not.

Reply:
Done, update with if else